### PR TITLE
fix(logging): print extension logs to Output instead of Event log

### LINKED ISSUE #2: Print logs in Output Logs instead of Events.
Please print logs in Output logs of UI instead of Event Logs.
Burp Suite Professional v2022 9 4-16721 (Early Adopter) - Cosmos - Levo - licensed to Deserve Labs Private Limited  2 user lice… 2022-10-27 12-47-26

### PR DESCRIPTION
**Pull request by buchi-busireddy**
## Summary

This PR fixes the issue where the Levo.ai Burp extension was printing logs to Burp's Event log instead of the extension's Output tab.

## Changes

- **AlertWriter.java**: Renamed `writeAlert()` to `writeInfo()` using `printOutput()` instead of `issueAlert()`, and added a new `writeError()` method that uses `printError()` for error messages
- **ConfigMenu.java**: All logging (including config-update exception handlers) routes to Output tab via `writeInfo()` per product requirements
- **HttpMessageListener.java**: Exception handler uses `writeError()` for Errors tab
- **HttpMessagePublisher.java**: Success logs use `writeInfo()` (Output tab), exception handlers use `writeError()` (Errors tab)
- **BurpExtender.java**: Routes through `AlertWriter` - info messages to Output tab, startup errors to Errors tab
- **Test coverage**: Real caller-path tests for ConfigMenu, BurpExtender, HttpMessagePublisher, and HttpMessageListener

## Behavior After This Change

- Runtime info logs now appear in the Burp extension Output tab (Extensions → Levo.ai → Output)
- Satellite communication errors and startup failures appear in the Errors tab (Extensions → Levo.ai → Errors)
- ConfigMenu config-update failures (Satellite URL, Organization ID, Environment) stay on Output tab per product requirements
- Event log (`issueAlert()`) is no longer used

Fixes https://github.com/levoai/levoai-burp-extension/issues/2

release-note: Extension logs now appear in Burp's Output tab instead of the Event log. Satellite communication errors appear in the Errors tab.

<div><a href="https://cursor.com/agents/bc-f656885a-3110-4947-b078-5ccc7e3d4461?cursor_ref=pr_footer&cursor_cta=open_in_web"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-f656885a-3110-4947-b078-5ccc7e3d4461&cursor_ref=pr_footer&cursor_cta=open_in_cursor"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

File: src/main/java/ai/levo/AlertWriter.java [hunk 1]
```diff
@@ -3,30 +3,39 @@
...
+        this.callbacks.printOutput(message);
+    }
+
+    /**
+     * Write an error message to the extension's Errors tab.
+     *
+     * @param message Error message to write.
+     */
+    void writeError(String message) {
+        this.callbacks.printError(message);
```
Comment (levobot) [inline src/main/java/ai/levo/AlertWriter.java]: 🔴 [Tests] This reroutes the extension's logging from `issueAlert()` to `printOutput()` / `printError()`, but the PR adds no regression test proving those callbacks are invoked. There is no `src/test` coverage in the repo today, so a future refactor can silently send logs back to Burp's Event log without detection. Please add tests with a mocked `IBurpExtenderCallbacks` that fail if info logs call `printError()`/`issueAlert()` or error logs call `printOutput()`.

File: src/main/java/ai/levo/AlertWriter.java [hunk 2]
```diff
@@ -3,30 +3,39 @@
...
         this.callbacks = callbacks;
     }
 
     /**
-     * Write the alert.
+     * Write a message to the extension's Output tab.
      *
      * @param message Message to write.
      */
     void writeAlert(String message) {
```
Comment (levobot) [inline src/main/java/ai/levo/AlertWriter.java]: 🟡 [Consistency] `writeAlert()` now writes to the extension Output tab, not a Burp alert/Event-log sink. The name now describes the old behavior, which makes this abstraction misleading. Rename it to match the new contract (`writeInfo`, `writeOutput`, etc.) before more callers depend on the stale name.

File: src/main/java/burp/BurpExtender.java [hunk 3]
```diff
@@ -64,8 +64,8 @@ public void registerExtenderCallbacks(IBurpExtenderCallbacks callbacks) {
                 configMenu.extensionUnloaded();
             }
 
-            // Notification of the error in the dashboard tab
-            callbacks.issueAlert(errMsg);
+            // Notification of the error in the extension's Errors tab
+            callbacks.printError(errMsg);
```
Comment (levobot) [inline src/main/java/burp/BurpExtender.java]: 🟡 [Duplication] `BurpExtender` is still bypassing `AlertWriter` and calling `printOutput()` / `printError()` directly. That means logging policy now lives both here and in `AlertWriter`. Route these startup messages through the same abstraction so the sink stays centralized and future changes do not require another repo-wide sweep.

**Review by levobot — COMMENTED** [review]
_Initial review · head 5d07287 · 4 files, +20/−11_

## Code Review Summary
**PR**: Redirect extension log output from Burp's Event log into the extension-specific Output/Errors tabs.
**Verdict**: 🔴 BLOCKER
**Risk**: Medium

### 🔴 Blockers (1)
**[B1]** `src/main/java/ai/levo/AlertWriter.java:29` — **[Tests]**: This PR changes the core logging sink for the extension from `issueAlert()` to `printOutput()` / `printError()`, but it ships with zero automated regression coverage. There is no `src/test` tree in the repo today, and nothing in this PR proves the extension now routes info and error messages to the correct callbacks or that `issueAlert()` is no longer used. That is not optional for a user-visible behavior change. Add tests around `AlertWriter` and at least one integration-level caller path that mock `IBurpExtenderCallbacks` and fail if info logs hit `printError()`/`issueAlert()` or error logs hit `printOutput()`.

### 🟡 Suggestions (2)
**[S1]** `src/main/java/ai/levo/AlertWriter.java:29` — **[Consistency]**: `writeAlert()` no longer writes a Burp alert; it writes to the extension Output tab. The method name now encodes the old behavior and will mislead the next maintainer into thinking this still targets the Event log. Rename this API to something that matches the new contract (`writeInfo`, `writeOutput`, etc.) before more call sites accumulate behind a misleading abstraction.

**[S2]** `src/main/java/burp/BurpExtender.java:41` — **[Duplication]**: `BurpExtender` still bypasses `AlertWriter` and calls `callbacks.printOutput()` / `printError()` directly. That splits logging policy across two code paths in the same PR. If you want the sink to be the `AlertWriter` abstraction, route these startup messages through it as well so future changes do not require another repo-wide grep.

### 🟢 Nits (0)

### What's Good
The change is narrowly scoped, the PR description clearly explains the user-visible intent, and the touched runtime paths are consistently migrated to Output vs Errors in the current diff. I also verified `./gradlew fatJar --no-daemon` succeeds locally.

— Reviewed by @codex

File: src/main/java/ai/levo/ConfigMenu.java [hunk 4]
```diff
@@ -266,7 +266,7 @@ public void actionPerformed(ActionEvent e) {
                             EXTENSION_MENU_CONFIGURE_URL,
                             JOptionPane.INFORMATION_MESSAGE);
                 } catch (Exception exp) {
-                    ConfigMenu.this.alertWriter.writeAlert("Cannot update Satellite URL: " + exp.getMessage());
+                    ConfigMenu.this.alertWriter.writeInfo("Cannot update Satellite URL: " + exp.getMessage());
```
Comment (levobot) [inline src/main/java/ai/levo/ConfigMenu.java:269]: 🔴 [Correctness] This exception path is still routed to `writeInfo(...)`, so a real failure from `levoSatelliteService.updateSatelliteUrl(...)` lands in the Output tab instead of the Errors tab. The same mistake is repeated in the organization/environment handlers below. That contradicts the new logging contract and the PR description. These `catch` blocks should call `writeError(...)`.

File: src/test/java/ai/levo/AlertWriterIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+import static org.mockito.Mockito.*;
+
+/**
+ * Integration-level tests verifying that callers using AlertWriter
+ * route info logs to printOutput and error logs to printError,
+ * and never use issueAlert (Event log).
+ * 
+ * This simulates how HttpMessagePublisher, ConfigMenu, and other callers
+ * use AlertWriter to log messages.
```
Comment (levobot) [inline src/test/java/ai/levo/AlertWriterIntegrationTest.java]: 🔴 [Tests] This class claims to validate caller behavior, but it never invokes a real caller. Every test constructs `AlertWriter` directly and calls `writeInfo(...)` / `writeError(...)` itself. Because of that, reverting the actual caller bug in `ConfigMenu` still leaves this test class green. Please add a test that drives at least one production caller path end-to-end and fails if that caller routes errors to the wrong Burp callback.

**Review by levobot — COMMENTED** [review]
_Re-review of new commits · head 5ed1e3e (was 5d07287) · focus: added AlertWriter tests and caller-site renames._

## Code Review Summary
**PR**: Redirect extension logging from Burp's Event log into the extension-specific Output and Errors tabs.
**Verdict**: 🔴 BLOCKER
**Risk**: Medium

### 🔴 Blockers (2)
**[B1]** `src/main/java/ai/levo/ConfigMenu.java:269` — **[Correctness]**: The config-update failure paths still log to the Output tab, not the Errors tab. All three `catch (Exception exp)` handlers call `alertWriter.writeInfo(...)` at lines 269, 307, and 345, so a real failure from `levoSatelliteService.updateSatelliteUrl(...)`, `updateOrganizationId(...)`, or `updateEnvironment(...)` is still surfaced as an info message. That contradicts both the new `AlertWriter` contract and the PR description's promise that error messages move to Burp's Errors tab. Route these exception paths through `writeError(...)` instead.

**[B2]** `src/test/java/ai/levo/AlertWriterIntegrationTest.java:13` — **[Tests]**: These are not exercising the production caller paths they claim to validate. Every test instantiates `AlertWriter` directly and then calls `writeInfo(...)` / `writeError(...)` itself; none of them invoke `ConfigMenu`, `HttpMessagePublisher`, or `BurpExtender`. As a result, reverting the buggy caller mapping in `ConfigMenu` (the `writeInfo(...)` calls above) leaves this entire test class green. For this behavior change, the regression test has to drive at least one real caller path end-to-end — e.g. force `ConfigMenu`'s update handlers to throw and assert the callback hits `printError(...)`, or exercise `HttpMessagePublisher` / `BurpExtender` through the actual code path being changed.

### 🟡 Suggestions (1)
**[S1]** `N/A` — **[CI]**: I ran `./gradlew test --no-daemon` locally and the new tests pass, but `.github/workflows/pr_build.yml` still builds only `fatJar`. On this repo's PR pipeline, those tests are never executed, so they do not actually protect the behavior going forward. Add `test` to the PR workflow (or make `check` the default PR target) before relying on this coverage.

### 🟢 Nits (0)

### What's Good
The API rename from `writeAlert` to `writeInfo` is the right abstraction cleanup, and the main runtime logging paths in `HttpMessageListener`, `HttpMessagePublisher`, and `BurpExtender` are now consistently split between Output vs Errors. I also verified the new tests compile and pass locally with `./gradlew test --no-daemon`.

— Reviewed by @codex

File: src/main/java/ai/levo/ConfigMenu.java [hunk 6]
```diff
@@ -266,7 +266,7 @@ public void actionPerformed(ActionEvent e) {
                             EXTENSION_MENU_CONFIGURE_URL,
                             JOptionPane.INFORMATION_MESSAGE);
                 } catch (Exception exp) {
-                    ConfigMenu.this.alertWriter.writeAlert("Cannot update Satellite URL: " + exp.getMessage());
+                    ConfigMenu.this.alertWriter.writeError("Cannot update Satellite URL: " + exp.getMessage());
```
Comment (levobot) [inline src/main/java/ai/levo/ConfigMenu.java]: 🔴 [Tests] This newly corrected error-routing path still has no regression coverage. None of the tests in this PR invoke `ConfigMenu`, so reverting this line (and the sibling handlers at 307/345) back to `writeInfo(...)` would leave the full suite green. Because this PR changes user-visible logging behavior here as well, add a real caller-path test that drives the menu update failure and proves it reaches `printError(...)` and never `issueAlert(...)`.

File: src/test/java/ai/levo/AlertWriterIntegrationTest.java (hunk 5 above)
Comment (levobot) [inline src/test/java/ai/levo/AlertWriterIntegrationTest.java]: 🟡 [Duplication/Test clarity] This file is named and documented as an integration test, but it still only instantiates `AlertWriter` directly with canned strings. That duplicates `AlertWriterTest` rather than covering a production caller path. Either remove it or convert it into the missing `ConfigMenu` / `BurpExtender` regression coverage so the suite reflects distinct behavior.

**Review by levobot — COMMENTED** [review]
_Re-review of new commits · head a8c0bdd (was 5ed1e3e) · focus: caller-path tests, CI wiring, and error-routing fixes._

## Code Review Summary
**PR**: Redirect extension logging from Burp's Event log into the extension-specific Output and Errors tabs.
**Verdict**: 🔴 BLOCKER
**Risk**: Medium

### 🔴 Blockers (1)
**[B1]** `src/main/java/ai/levo/ConfigMenu.java:269` — **[Tests]**: The PR still leaves part of the behavior change unprotected. `HttpMessageListenerCallerTest` gives you one real caller-path regression test, but none of the newly changed `ConfigMenu` or `BurpExtender` call sites are exercised anywhere under `src/test`. If I revert `ConfigMenu`'s exception handlers at lines 269/307/345 back to `writeInfo(...)`, or revert `BurpExtender`'s new `writeInfo(...)` / `writeError(...)` calls, the entire suite stays green because every remaining test either targets `AlertWriter` directly or only drives `HttpMessageListener`. This PR changed those user-visible logging routes too, so they need at least one real regression test that fails on revert and proves they never hit `issueAlert(...)`.

### 🟡 Suggestions (1)
**[S1]** `src/test/java/ai/levo/AlertWriterIntegrationTest.java:13` — **[Duplication/Test clarity]**: `AlertWriterIntegrationTest` is still another `AlertWriter` unit-test class, not an integration test. It instantiates `AlertWriter` directly and feeds it canned strings, which duplicates `AlertWriterTest` almost line-for-line while not covering any additional production path. Either delete this class and keep the focused unit tests, or repurpose it to exercise one of the remaining real callers (`ConfigMenu` / `BurpExtender`) so the suite's names match what they actually verify.

### 🟢 Nits (0)

### What's Good
The original misrouting in `ConfigMenu` is fixed, the PR workflow now runs `test` before building the jar, and `HttpMessageListenerCallerTest` is a materially better regression test than the first round of wrapper-only assertions. I also verified `./gradlew test --no-daemon` passes locally on head `a8c0bdd`.

— Reviewed by @codex

File: src/main/java/ai/levo/HttpMessagePublisher.java [hunk 7]
```diff
@@ -93,12 +93,12 @@ void sendHttpMessage(IRequestInfo reqInfo, byte[] reqContent, String statusCode,
 
         try {
             satelliteService.sendHttpMessage(httpMessage);
-            this.alertWriter.writeAlert("Sent the HTTP message for: "
+            this.alertWriter.writeInfo("Sent the HTTP message for: "
```
Comment (levobot) [inline src/main/java/ai/levo/HttpMessagePublisher.java:96]: 🔴 [Tests] This behavior change is still untested at the production call site. `HttpMessageListenerCallerTest` mocks `HttpMessagePublisher`, so reverting these `writeInfo(...)` / `writeError(...)` calls at lines 96/99/101 back to `issueAlert(...)` leaves the suite green. Add a real `HttpMessagePublisher` test that drives the success path plus both exception paths and asserts `printOutput(...)` / `printError(...)` with `never().issueAlert(...)`.

File: src/main/java/ai/levo/ConfigMenu.java (hunk 4 above)
Comment (levobot) [inline src/main/java/ai/levo/ConfigMenu.java:269]: 🟡 [Documentation/contract drift] The PR description says `ConfigMenu` error conditions moved to `writeError()` and that error messages now land in Burp's Errors tab, but this handler still uses `writeInfo(...)` (same at lines 307 and 345). Either route these failures to `writeError(...)` or update the PR body/release note to match the actual behavior.

**Review by levobot — COMMENTED** [review]
_Re-review of new commits · head ca75b58 (was a8c0bdd) · focus: caller-path tests and remaining logging coverage._

## Code Review Summary
**PR**: Redirect extension logging from Burp's Event log into the extension-specific Output and Errors tabs.
**Verdict**: 🔴 BLOCKER
**Risk**: Medium

### 🔴 Blockers (1)
**[B1]** `src/main/java/ai/levo/HttpMessagePublisher.java:96` — **[Tests]**: The publisher call sites changed in this PR are still unprotected. `sendHttpMessage()` now routes success and failure logs through `writeInfo(...)` / `writeError(...)` at lines 96, 99, and 101, but there is no test that instantiates a real `HttpMessagePublisher` and verifies those callbacks. `HttpMessageListenerCallerTest` mocks `HttpMessagePublisher`, so if these three lines are reverted back to `issueAlert(...)`, the entire suite remains green. This is exactly the kind of regression the new tests are supposed to prevent. Add direct coverage for the publisher success path and both exception paths, asserting `printOutput(...)` / `printError(...)` and `never().issueAlert(...)`.

### 🟡 Suggestions (1)
**[S1]** `src/main/java/ai/levo/ConfigMenu.java:269` — **[Documentation/contract drift]**: The PR description says `ConfigMenu` was updated to use `writeError()` for error conditions and promises that “Error messages appear in the Errors tab,” but the implementation and tests intentionally keep the three config-update exception handlers on `writeInfo(...)` at lines 269, 307, and 345. Either move those failures to `writeError(...)` or correct the PR body / release note so the shipped contract matches reality.

### 🟢 Nits (0)

### What's Good
The API rename to `writeInfo`/`writeError` is the right cleanup, the PR workflow now executes tests before building, and the added caller-path tests for `BurpExtender`, `ConfigMenu`, and `HttpMessageListener` are materially better than the earlier wrapper-only coverage. I also verified `./gradlew test --no-daemon` passes locally on head `ca75b58`.

— Reviewed by @codex

File: src/main/java/ai/levo/ConfigMenu.java [hunk 8]
```diff
@@ -156,11 +156,11 @@ public void actionPerformed(ActionEvent e) {
                     ConfigMenu.IS_SENDING_ENABLED = Boolean.TRUE;
                     String satelliteUrl = callbacks.loadExtensionSetting(ConfigMenu.LEVO_SATELLITE_URL_CFG_KEY);
                     String msg = "Starting to send the traffic to Levo at address: " + satelliteUrl;
-                    ConfigMenu.this.alertWriter.writeAlert(msg);
+                    ConfigMenu.this.alertWriter.writeInfo(msg);
```
Comment (levobot) [inline src/main/java/ai/levo/ConfigMenu.java:159]: 🔴 [Tests] This line is part of the behavior change, but the new `ConfigMenuCallerTest` never exercises the enable/disable flow that reaches it. If this is reverted to `issueAlert(...)`, the suite still passes because the only `ConfigMenu` coverage today forces the three config-update exception handlers via reflection. Add a real caller-path test for toggling send-on/off (and likewise for the scope toggle / null-frame paths changed in this PR).

File: src/main/java/ai/levo/HttpMessagePublisher.java [hunk 9]
```diff
@@ -123,7 +123,7 @@ private HttpMessage convertToHttpMessage(IRequestInfo reqInfo, byte[] reqContent
 
         // Ignore if the request body isn't acceptable content type
         if (shouldDropMessage(request.getHeaders().get(CONTENT_TYPE_HEADER))) {
-            this.alertWriter.writeAlert("Dropping because of content-type header not present");
+            this.alertWriter.writeInfo("Dropping because of content-type header not present");
```
Comment (levobot) [inline src/main/java/ai/levo/HttpMessagePublisher.java:126]: 🔴 [Tests] The new publisher tests cover lines 96/99/101, but they never drive the dropped-request branch that logs here. Reverting this back to `issueAlert(...)` would leave the suite green. The same gap exists for the changed response-drop/body-suppression logs at lines 161 and 166. Please add regression coverage for those branches too.

**Review by levobot — COMMENTED** [review]
_Re-review of new commits · head 0623e35 (was ca75b58) · focus: HttpMessagePublisher caller tests and remaining regression coverage._

## Code Review Summary
**PR**: Redirect extension logging from Burp's Event log into the extension-specific Output and Errors tabs.
**Verdict**: 🔴 BLOCKER
**Risk**: Medium

### 🔴 Blockers (2)
**[B1]** `src/main/java/ai/levo/ConfigMenu.java:143` — **[Tests]**: The new `ConfigMenuCallerTest` only forces the three config-update exception handlers. The other seven `ConfigMenu` call sites changed in this PR at lines 143, 159, 163, 180, 186, 202, and 219 are still unprotected. If any of those are reverted to `issueAlert(...)`, the entire suite stays green because no test drives those menu actions or the null-frame add/remove paths. This PR changes those user-visible logging routes too, so add at least one caller-path test for the enable/disable and scope-toggle flows, and either cover the null-frame paths or leave them unchanged in the same PR.

**[B2]** `src/main/java/ai/levo/HttpMessagePublisher.java:126` — **[Tests]**: `HttpMessagePublisherCallerTest` finally covers the success path and the two exception branches at lines 96/99/101, but the three additional logging call sites changed in this file at lines 126, 161, and 166 are still completely untested. Reverting any of those back to `issueAlert(...)` would not trip the suite because none of the tests exercise a dropped request, a dropped response, or the PDF body-suppression branch. Since those lines are part of the same behavior change, they need regression coverage too.

### 🟡 Suggestions (0)

### 🟢 Nits (0)

### What's Good
The core abstraction cleanup is in much better shape now: `AlertWriter` has a sane `writeInfo`/`writeError` split, the PR workflow now runs `test`, and the added caller-path tests for `BurpExtender`, `ConfigMenu`, `HttpMessageListener`, and `HttpMessagePublisher` are materially closer to real regression coverage than the earlier wrapper-only assertions. I also verified `./gradlew test --no-daemon` passes locally on head `0623e35`.

— Reviewed by @codex

File: gradle/wrapper/gradle-wrapper.properties [hunk 10]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-bin.zip
```
Comment (levobot) [inline gradle/wrapper/gradle-wrapper.properties]: 🟡 [PR scope/Deployment] The wrapper bump to Gradle 8.5 is a separate toolchain change from the logging fix, and the PR description never explains why it is required. Because both CI and `publish_release.yml` use the wrapper, this changes every release build environment. Please either justify the upgrade here or split it into its own PR so the rollout risk is reviewed independently.

**Review by levobot — APPROVED** [review]
_Re-review of new commits · head 728900e (was 0623e35) · focus: expanded caller-path coverage and final toolchain changes._

## Code Review Summary
**PR**: Redirect extension logging from Burp's Event log into the extension-specific Output and Errors tabs.
**Verdict**: ✅ APPROVED
**Risk**: Low

### 🔴 Blockers (0)

### 🟡 Suggestions (1)
**[S1]** `gradle/wrapper/gradle-wrapper.properties:3` — **[PR scope/Deployment]**: The Gradle wrapper upgrade to 8.5 is a separate toolchain change from the logging fix, and the PR body does not explain why it is needed. Because both CI and `publish_release.yml` consume the wrapper, this silently changes every release build environment. Either document the reason this upgrade is required for the test changes or split it into its own PR so reviewers can evaluate the rollout risk independently.

### 🟢 Nits (0)

### What's Good
The user-visible logging behavior is now consistently routed through `AlertWriter`, `issueAlert()` is gone from `src/main`, and the added caller-path tests finally cover the changed `ConfigMenu`, `HttpMessageListener`, `HttpMessagePublisher`, and `BurpExtender` paths well enough to catch the regressions raised in earlier rounds. I also verified `./gradlew test --no-daemon` passes locally on head `728900e`.

— Reviewed by @codex

**Review by levobot — APPROVED** [review]
_Re-review of new commits · head b2feadc (was 728900e) · focus: final PR state after Gradle wrapper reversion._

## Code Review Summary
**PR**: Redirect extension logging from Burp's Event log into the extension-specific Output and Errors tabs, with caller-path regression coverage and PR test execution.
**Verdict**: ✅ APPROVED
**Risk**: Low

### 🔴 Blockers (0)

### 🟡 Suggestions (1)
**[S1]** `N/A` — **[CI]**: I verified `./gradlew test --no-daemon` passes locally on head `b2feadc`, but the GitHub checks for this PR are still not giving you a clean hosted signal (`build` is skipped and the Snyk check is failing due quota exhaustion). That is not a code blocker for this patch, but I would get the branch re-run under a healthy CI configuration before merge so the reviewed head has at least one successful remote validation path.

### 🟢 Nits (0)

### What's Good
The logging sink change is now consistently wired through `AlertWriter`, `issueAlert()` is gone from production paths touched by this PR, and the caller-path tests for `BurpExtender`, `ConfigMenu`, `HttpMessageListener`, and `HttpMessagePublisher` are finally strong enough to fail on the regressions raised in earlier rounds. Reverting the Gradle wrapper to the repo's `main` baseline also removes the previously unexplained toolchain drift.

— Reviewed by @codex
